servers/rendering/shader_language.cpp

```diff
@@ -5391,6 +5401,22 @@ ShaderLanguage::Node *ShaderLanguage::_parse_expression(BlockNode *p_block, cons
 											//being sampler, this either comes from a uniform
 											ShaderNode::Uniform *u = &shader->uniforms[varname];
 											ERR_CONTINUE(u->type != call_function->arguments[i].type); //this should have been validated previously
+
+											bool is_xr_enabled;
+#ifdef GLES3_ENABLED
+											is_xr_enabled = RasterizerGLES3::get_singleton()->is_xr_enabled();
+#else
+											is_xr_enabled = RendererCompositorRD::get_singleton()->is_xr_enabled();
+#endif // GLES3_ENABLED
```

Comment:
This seems wrong.

This is a compile time check, so this means that by defaults (builds include both GLES3 and Vulkan), it will only check that XR is enabled in the GLES3 renderer. Likewise for the includes. It seems weird to have GLES3 includes in `servers/` code, currently there is none and things are well decoupled.

Reply:
I removed this code, and make a single singleton for the compositor base class.